### PR TITLE
Update Dockerfile to allow for config overrides

### PR DESCRIPTION
**Pull request by devkelley**
## Motivation and Context

Allows for an image created by the Dockerfile to be run with a configuration file that can be passed in when running the container.

## Description

This PR does the following:
- Adds a startup script used by the container to start the service. This script also copies any configuration passed into the container.
- Modified the Dockerfile to copy configuration from the mounted configuration directory to the service's configuration directory.
- Added how to run the container image with configuration at runtime for docker and podman

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -62,6 +62,14 @@ RUN adduser \
     --no-create-home \
     --uid "${UID}" \
     appuser
+
+# Create and add user ownership to config directory.
+RUN mkdir -p /sdv/.agemo/config
+RUN chown appuser /sdv/.agemo/config
```
Comment (ashbeitz) [inline Dockerfile:68]: Where does appuser come from?  Is it a special user that we need to setup?

File: README.md [hunk 2]
```diff
@@ -197,9 +197,9 @@ see more full featured examples in
 
 ## Running in a Container
 
-Below are the steps for running the service in a container. Note that the configuration files used
-by the containerized service are cloned from [.agemo/config](.agemo/config/) defined in the
-project's root.
+Below are the steps for running the service in a container. Note that the default configuration
+files used by the containerized service are cloned from [.agemo/config](.agemo/config/) defined in
```
Comment (ashbeitz) [inline README.md:201]: A change to Agemo could cause this to break.  I assume that we are locked in to a specific version.

File: container_startup.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
```
Comment (ashbeitz) [inline container_startup.sh:2]: Please add a "set -e"

**Review by ashbeitz — COMMENTED** [review]
A few comments.  Please take a look.

File: README.md [hunk 4]
```diff
@@ -285,6 +302,23 @@ root directory:
     podman ps -f ancestor=localhost/pub_sub_service:latest --format="{{.Names}}" | xargs podman stop
     ```
 
+#### Running in Podman with overrided configuration
```
Comment (jorchiu) [inline README.md:305]: Please change `overrided` to `overridden`

File: Dockerfile (hunk 1 above)
Comment (devkelley) [reply Dockerfile:68]: appuser is setup in the lines above this. It is just a non-privileged user for running the service

File: container_startup.sh (hunk 3 above)
Comment (devkelley) [reply container_startup.sh:2]: added

File: README.md (hunk 4 above)
Comment (devkelley) [reply README.md]: updated here and elsewhere

File: README.md (hunk 2 above)
Comment (devkelley) [reply README.md:201]: Talked with ash offline, there was a bit of a miscommunication. No action needed here.

**Review by jorchiu — APPROVED** [review]

**Review by ashbeitz — APPROVED** [review]
Approved.

**Review by ladatz — APPROVED** [review]

**Comment by devkelley** [comment]
markdown-link-check workflow is having issues. It is marked as failing but have manually verified all 'broken' links are valid.

**Review by ladatz — APPROVED** [review]

**Pull request merged**